packages/connect/src/data/firmwareInfo.ts

```diff
@@ -44,220 +71,228 @@ export const getOnlineReleases = async (internalModel: DeviceModelInternal) => {
+    firmwareReleasesConfig = config.releases;
+    firmwareIntermediaryReleasesConfig = config.intermediaries;
 
-    if (features.bootloader_mode) {
-        // the problem with bootloader is that we see only bootloader and not firmware version
-        // and multiple releases may share same bootloader version. we really can not tell that
-        // the versions that are installable are newer. so...
-        if (features.firmware_present && features.major_version === 1) {
-            // return null signaling that we don't really know, but only if some firmware
-            // is already installed!
-            return null;
-        }
-        if (features.firmware_present && features.major_version === 2) {
-            // little different situation is with model 2, where in bootloader (and with some fw installed)
-            // we actually know the firmware version
-            return releases2.filter(r =>
-                versionUtils.isNewer(r.version, [
-                    features.fw_major,
-                    features.fw_minor,
-                    features.fw_patch,
-                ]),
-            );
-        }
-
-        // for fresh devices, we can assume that all releases are actually "new"
-        return releases2;
-    }
-
-    // otherwise we are in firmware mode and because each release in releases list has
-    // version higher than the previous one, we can filter out the version that is already
-    // installed and show only what's new!
-    return releases2.filter(r =>
-        versionUtils.isNewer(r.version, [
-            features.major_version,
-            features.minor_version,
-            features.patch_version,
-        ]),
-    );
+    return {
+        releases,
+        intermediaryReleases: firmwareIntermediaryReleasesConfig,
+    };
 };
 
-const isNewer = (release: FirmwareRelease, features: StrictFeatures) => {
-    if (features.major_version === 1 && features.bootloader_mode) {
-        return null;
-    }
+const getCurrentFirmwareVersion = (features: Features): VersionArray => {
+    const {
+        bootloader_mode,
+        major_version,
+        minor_version,
+        patch_version,
+        fw_major,
+        fw_minor,
+        fw_patch,
+    } = features;
 
-    return versionUtils.isNewer(release.version, [
-        features.major_version,
-        features.minor_version,
-        features.patch_version,
-    ]);
+    const firmwareVersionInBootloaderMode = [fw_major, fw_minor, fw_patch] as VersionArray;
+    const version = [major_version, minor_version, patch_version] as VersionArray;
+
+    return bootloader_mode && firmwareVersionInBootloaderMode
+        ? firmwareVersionInBootloaderMode
+        : version;
 };
 
-const isRequired = (changelog: ReturnType<typeof getChangelog>) => {
-    if (!changelog || !changelog.length) return null;
+const getCurrentBootloaderVersion = (features: Features): VersionArray => {
```

Comment:
personally, I don't like very much creating a method with a very specific name `getCurrentBootloaderVersion` that gives you current bootloader version only if called under specific circumstances (features.bootloader_mode === true)

At the first glance I thought that this would give me bootloader version of any connected device and went to create a screenshot to prove that this is not true only later to notice that it was not meant to be used like this.

<img width="1080" alt="image" src="https://github.com/user-attachments/assets/41b0802e-90db-47e3-8abc-47168c224275" />

Reply:
if it is meant to be specif I would call it `getCurrentBootloaderVersionForDeviceInBootloaderMode`

Reply:
You are right, but thinking about it. My goal separating into functions was to make it more clear because before there was a smart function that was returning you version but was not clear if bootloader or firmware version.

I am thinking to get back to the concept of using a smarter function that will handled all the functions. Something maybe called `getCurrentDeviceVersions` that will return an object:

```
{
  firmwareVersion: ...
  bootloaderVersion: ...
}
```
And depending of if it is run in bootloader/firmware mode it will return you the version or undefined or whatever it is, and well commented.

WDYT?

Reply:
@mroz22 what about https://github.com/trezor/trezor-suite/pull/18821/commits/dbe8cff129803fd3f92164215064c32ea3019d4d  ?

I am consider if I should consider the situation `Some old version of T1B1 do not report FW version in bootloader mode` in the typescript types and check for that in the function, or is it enough to ignore it there and just check for it in https://github.com/trezor/trezor-suite/blob/develop/packages/connect/src/data/firmwareInfo.ts#L244  since it only happens in very old versions of T1.